modules/cnv-connecting-resource-bridge-network.adoc

```diff
@@ -0,0 +1,85 @@
+// Module included in the following assemblies:
+//
+// * cnv/cnv-attaching-vm-multiple-networks.adoc
+
+[id="connecting-resource-bridge-network-{context}"]
+= Connecting a resource to a bridge-based network
+
+As a network administrator, you can configure a NetworkAttachmentDefinition
+of type `cnv-bridge` to provide Layer-2 networking to Pods and virtual machines.
+
+.Prerequisites
+
+* {ProductName} 2.0 or newer
+* A Linux bridge must be configured and attached to the correct
+Network Interface Card on every node.
+* If you use VLANs, `vlan_filtering` must be enabled on the bridge.
+* The NIC must be tagged to all relevant VLANs.
+** For example: `bridge vlan add dev bond0 vid 1-4095 master`
+
+.Procedure
+
+. Create a new file for the NetworkAttachmentDefinition in any local directory.
+The file must have the following contents, modified to match your
+configuration:
++
+[source,yaml]
+----
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: a-bridge-network
+spec:
+  config: '{
+    "cniVersion": "0.3.0",
+    "name": "a-bridge-network", <1>
+    "type": "cnv-bridge", <2>
+    "bridge": "br0", <3>
+    "isGateway": true,
+    "ipam": {}
+}'
+----
+<1> The `name` value is part of the annotation you will use in the next step.
+<2> The actual name of the Container Network Interface (CNI) plug-in that provides
+the network for this NetworkAttachmentDefinition. Do not change this field unless
+you want to use a different CNI.
+<3> You must substitute the actual name of the bridge, if it is not `br0`.
+
+. Edit the configuration file of a resource that you want to connect to the
+bridge network:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example-pod
+  annotations:
+    k8s.v1.cni.cncf.io/networks: a-bridge-network <1>
```

Comment:
The pod/vmi can also be scheduled on a node where the bridge used in a NetworkAttachmentDefinition is available.

To mark the nodes with available bridges the bridge marker daemon (https://github.com/kubevirt/bridge-marker/) must be running. The daemon can be installed using the Network Addons Operator (https://github.com/kubevirt/cluster-network-addons-operator).
In order for a vmi to be scheduled on available nodes, the NetworkAttachmentDefinition must have an annotation marking that it requires the given bridge: 
  metadata.annotations:     k8s.v1.cni.cncf.io/resourceName: bridge-cni.network.kubevirt.io/<bridge name>
For example, to mark vmi's with a NetworkAttachmentDefinition to to be scheduled only on nodes where a bridge named "br1" is present, the network attachment defintion would look as follows:
[source,yaml]
----
apiVersion: "k8s.cni.cncf.io/v1"
kind: NetworkAttachmentDefinition
metadata:
  name: a-bridge-network
  annotations:
    k8s.v1.cni.cncf.io/resourceName: bridge-cni.network.kubevirt.io/br1
spec:
  config: '{
    "cniVersion": "0.3.0",
    "name": "a-bridge-network", 
    "type": "cnv-bridge",
    "bridge": "br0", 
    "isGateway": true,
    "ipam": {}
}'
----

To mark a pod as requiring the bridge, the pod would need to have a resouce specified, for example:

[source,yaml]
----
apiVersion: v1
kind: Pod
metadata:
  name: a-name
  namespace: default
spec:
  containers:
  - name: a-name
    image: docker.io/...
    resources:
      limits:
        bridge.network.kubevirt.io/zbr: "1"
      requests:
        bridge.network.kubevirt.io/zbr: "1"
...
----------------------------------

Reply:
Hey Marcin, thanks for the review and information! Can you clarify if the bridge marker daemon (and network addons operator) are "supported" in downstream CNV 2.0? I hadn't heard about those until now. 

@dankenigsberg - is adding info about this required for CNV-127?

(FYI, I can't directly tag you, Marcin - I think it's because you aren't in the OpenShift org)

Reply:
oh now I can comment here. @ousleyp you can learn more about the bridge marker in https://jira.coreos.com/browse/CNV-701 but I don't think it is required. all that we need to say: "if you wanna use it, add the annotation, and your VMs would run only on hosts that have this bridge connected."

Reply:
@dankenigsberg thanks for clarifying! How's this? https://github.com/openshift/openshift-docs/pull/14351/commits/fa6734e7bd7ba0f4159bb55e0eb26d62ef91e3e9